pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
+        self._names(var_name, extra_index,time_index, values, upper_bound, lower_bound, check_values_bounds=False)
+
+        self._add_elements(var_name, extra_index=extra_index, time_index=time_index)
+
+        self._names(var_name, extra_index,time_index, values, upper_bound, lower_bound, check_values_bounds=True)
+
+        if values:
+            # this dictionary keys are special set, values are its value
+            self.special_set_value = self._generate_dict(values)
+
+        if upper_bound:
+            if type(upper_bound) in [int, float]:
+                upper_bound = [upper_bound]*len(self.special_set)
+            self.upper_bound = self._generate_dict(upper_bound)
+            
+        if lower_bound:
+            if type(lower_bound) in [int, float]:
+                lower_bound = [lower_bound]*len(self.special_set)
+            self.lower_bound = self._generate_dict(lower_bound)
+
+        return self.special_set
+    
+    def update_values(self, values):
         """
-        Generate a dictionary, keys are the variable names, values are the indexes of this variable.
-        For e.g., name_and_index = {'C': ['CA', 'CB', 'CC']}
+        Used for updating values
+
         Parameters
         ---------
-        all_info: a dictionary, keys are measurement variable names,
-                values are a dictionary, keys are its extra index, values are its measuring time points
-                values are a list of measuring time point if there is no extra index for this measurement
-            Note: all_info can be the self.measurement_all_info, but does not have to be it.
+         values: a ``list`` containing values which has the same shape of flattened variables 
+            default choice is None, means there is no give nvalues 
         """
-        measurement_name = list(all_info.keys())
-        # a list of measurement extra indexes
-        measurement_extra_index = []
-        # check if the measurement has extra indexes
-        for i in measurement_name:
-            if type(all_info[i]) is dict:
-                index_list = list(all_info[i].keys())
-                measurement_extra_index.append(index_list)
-            elif type(all_info[i]) is list:
-                measurement_extra_index.append(None)
-        # a dictionary, keys are measurement names, values are a list of extra indexes
-        self.name_and_index = dict(zip(measurement_name, measurement_extra_index))
-
-    def _generate_flatten_name(self, measure_name_and_index):
+        self.special_set_value = self._generate_dict(values)
+        
+
+    def _generate_dict(self, values):
         """
-        Generate measurement flattened names
-        Parameters
-        ----------
-        measure_name_and_index: a dictionary, keys are measurement names, values are lists of extra indexes
+        Given a list of values, return a dictionary, keys are special set names. 
+        """
+        value_map = {}
```

Comment:
Need to refactor this logic when changing `add_variables` to support multiple sequential calls.

Reply:
We may decide this function is not needed.